src/R2_alg.jl

```diff
@@ -136,15 +135,16 @@ function R2(
 
     # define model
     φk(d) = dot(∇fk, d)
-    mk(d) = φk(d) + σk * dot(d, d) / 2 + ψ(d)
+    mk(d) = φk(d) + ψ(d)
 
     prox!(s, ψ, mν∇fk, ν)
     Complex_hist[k] += 1
     mks = mk(s)
-    ξ = hk - mks + max(1, abs(hk)) * 10 * eps()
+    Δm = hk - mks + max(1, abs(hk)) * 10 * eps()
+    ξ = Δm - σk * dot(s, s) / 2
     ξ > 0 || error("R2: prox-gradient step should produce a decrease but ξ = $(ξ)")
 
-    if sqrt(ξ) < ϵ
+    if sqrt(ξ) * σk < ϵ
```

Comment:
I don't think we need this, do we? The theory says we should stop based on σk * ξ(xk; σmax), but we don't know σmax, so we settled for just ξ(xk; σk) as a proxy.

Reply:
Confused - wasn't this suggested in slack?

Reply:
We floated the idea, but does it resolve the issues with FH taking many iterations with a only 1 inner iter? Or does it solve anything else? Currently, it's making a unit test fail.

Reply:
It seems to do a bit better from what I've observed but leans more towards making the inner solver use a lot of iterations/max out every time. So I'm not sure what is best here. We can just leave it.